v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsCustomShardIT.java

```diff
@@ -0,0 +1,296 @@
+    // needed to create separate metadata database
+    spannerMetadataDatabaseName = spannerMetadataResourceManager.getDatabaseId();
+  }
+
+  private void launchReaderDataflowJob() throws IOException {
+    // default parameters
+    Map<String, String> params =
+        new HashMap<>() {
+          {
+            put("sessionFilePath", getGcsFullPath("input/session.json"));
+            put("instanceId", spannerResourceManager.getInstanceId());
+            put("databaseId", spannerResourceManager.getDatabaseId());
+            put("spannerProjectId", PROJECT);
+            put("metadataDatabase", spannerMetadataResourceManager.getDatabaseId());
+            put("metadataInstance", spannerMetadataResourceManager.getInstanceId());
+            put("sourceShardsFilePath", getGcsFullPath("input/shard.json"));
+            put("changeStreamName", "allstream");
+            put("runIdentifier", "run1");
+            put("gcsOutputDirectory", getGcsFullPath("output"));
+            put("shardingCustomJarPath", getGcsFullPath("input/customShard.jar"));
+            put("shardingCustomClassName", "com.custom.CustomShardIdFetcherForIT");
+          }
+        };
+
+    // Construct template
+    String jobName = PipelineUtils.createJobName("rr-it");
+    // /-DunifiedWorker=true when using runner v2
+    PipelineLauncher.LaunchConfig.Builder options =
+        PipelineLauncher.LaunchConfig.builder(jobName, specPath);
+    options.setParameters(params);
+    options.addEnvironment("additionalExperiments", Collections.singletonList("use_runner_v2"));
+    // Run
+    jobInfo = launchTemplate(options, false);
+    assertThatPipeline(jobInfo).isRunning();
+  }
+
+  private void createSpannerDatabase() throws IOException {
+    spannerResourceManager =
+        SpannerResourceManager.builder("rr-main-" + testName, PROJECT, REGION)
+            .maybeUseStaticInstance()
+            .build(); // DB name is appended with prefix to avoid clashes
+    String ddl =
+        String.join(
+            " ", Resources.readLines(Resources.getResource(spannerDdl), StandardCharsets.UTF_8));
+    ddl = ddl.trim();
+    String[] ddls = ddl.split(";");
+    for (String d : ddls) {
+      if (!d.isBlank()) {
+        spannerResourceManager.executeDdlStatement(d);
+      }
+    }
+    spannerDatabaseName = spannerResourceManager.getDatabaseId();
+  }
+
+  private void createAndUploadJarToGcs() throws IOException, InterruptedException {
+    String[] commands = {"cd ../spanner-custom-shard", "mvn install"};
+
+    // Join the commands with && to execute them sequentially
+    String[] shellCommand = {"/bin/bash", "-c", String.join(" && ", commands)};
+
+    Process exec = Runtime.getRuntime().exec(shellCommand);
+
+    IORedirectUtil.redirectLinesLog(exec.getInputStream(), LOG);
+    IORedirectUtil.redirectLinesLog(exec.getErrorStream(), LOG);
+
+    if (exec.waitFor() != 0) {
+      throw new RuntimeException("Error staging template, check Maven logs.");
+    }
```

Comment:
Should we follow the same approach of having a `TestBaseIT` class like we have defined for `DataStreamToSpanner`? 

Do I repeat all of this instantiation logic if I need to write another IT for `SpannerChangeStreamToGcs*`?

Reply:
Yes, makes sense. Will follow similar approach of having TestBaseIT

Reply:
Added